save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/TopBarUrl.kt

```diff
@@ -0,0 +1,158 @@
+package com.saveourtool.save.frontend.utils
+
+import com.saveourtool.save.entities.benchmarks.BenchmarkCategoryEnum
+import com.saveourtool.save.validation.FrontendRoutes
+
+/**
+ * The class for analyzing url address and creating links in topBar
+ * @property href
+ */
+class TopBarUrl(val href: String) {
+    /**
+     * CurrentPath is the link that we put in buttons
+     */
+    var currentPath = "#"
+    private var circumstance: SituationUrlClassification = SituationUrlClassification.KEYWORD_PROCESS
+    private var processLastSegments = 0
+    private val sizeUrlSegments: Int = href.split("/").size
+
+    init {
+        findExclude(href)
+    }
+
+    /**
+     * The function is called to specify the link address in the button before creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlBeforeButton(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT, SituationUrlClassification.ORGANIZATION -> "#/${FrontendRoutes.PROJECTS.path}"
```

Comment:
I did not really get into what happens here, but this string looks weird. Is this what we really want?

Reply:
yes